fiftyone/core/expressions.py

```diff
@@ -763,10 +763,10 @@ def to_mongo(self, prefix=None):
 
 
 def _escape_regex_chars(str_or_strs):
-    # Must escape `]` and `-` because they have special meaning inside the `[]`
-    # that will be used in the replacement regex
-    regex_chars = "[\]{}()*+\-?.,\\^$|#"
-    _escape = lambda s: re.sub(r"([%s])" % "".join(regex_chars), r"\\\1", s)
+    # Must escape `[`, `]`, `-`, and `\` because they have special meaning
+    # inside the `[]` that will be used in the replacement regex
+    regex_chars = r"\[\]{}()*+\-?.,\\^$|#"
```

Comment:
Changing this to a raw string was necessary for `\` to be escaped - before, it was turning into a single backslash in the character set in the regex below, which (unnecessarily) escaped the `^` instead.
An alternative would be to use 4 backslashes instead of a raw string (`"\[\]{}()*+\-?.,\\\\^$|#"`), but that quickly becomes hard to keep track of, and parts of regexes should generally use raw strings anyway, like full regexes.